src/HackF5.UnitySpy/AssemblyImageFactory.cs

```diff
@@ -42,29 +42,48 @@ public static IAssemblyImage Create(int processId, string assemblyName = "Assemb
             var process = new ProcessFacade(processId);
             var monoModule = AssemblyImageFactory.GetMonoModule(process);
             var moduleDump = process.ReadModule(monoModule);
-            var rootDomainFunctionAddress = AssemblyImageFactory.GetRootDomainFunctionAddress(moduleDump, monoModule);
+            var rootDomainFunctionAddress = AssemblyImageFactory.GetRootDomainFunctionAddress(moduleDump, monoModule, process.Is64Bits);
 
             return AssemblyImageFactory.GetAssemblyImage(process, assemblyName, rootDomainFunctionAddress);
         }
 
-        private static AssemblyImage GetAssemblyImage(ProcessFacade process, string name, int rootDomainFunctionAddress)
+        private static AssemblyImage GetAssemblyImage(ProcessFacade process, string name, IntPtr rootDomainFunctionAddress)
         {
-            var domainAddress = process.ReadPtr((uint)rootDomainFunctionAddress + 1);
-            //// pointer to struct of type _MonoDomain
-            var domain = process.ReadPtr(domainAddress);
+
+            IntPtr domain;
+            if (process.Is64Bits)
+            {
+                // Offsets taken by decompiling the 64 bits version of mono-2.0-bdwgc.dll
+                // mov rax, [rip + 0x46ad39]
+                // ret
+                var offset = process.ReadInt32(rootDomainFunctionAddress + 3) + 7;
```

Comment:
Do you think you could add a more detailed explanation for both numbers, so that others can figure it out more easily in the future if things change?

Reply:
Done